crates/ruff_python_formatter/src/statement/stmt_function_def.rs

```diff
@@ -60,21 +62,40 @@ impl FormatNodeRule<StmtFunctionDef> for FormatStmtFunctionDef {
 
         let format_inner = format_with(|f: &mut PyFormatter| {
             write!(f, [item.parameters.format()])?;
+
             if let Some(return_annotation) = item.returns.as_ref() {
                 write!(f, [space(), text("->"), space()])?;
-                if return_annotation.is_tuple_expr() {
+
+                if empty_parameters(&item.parameters, f.context().source()) {
```

Comment:
This is meant to capture "Can the parameters break?". Like, semantically, it'd be nice if I could add a group around parameters and check if it breaks, but I don't think I can do that.

Reply:
I think I may be able to clean this up by using a new `Parenthesize` variant + using `maybe_parenthesize_expression` for both paths (this would also fix some other bugs not covered by this PR), though not sure how conservative we want to be with adding `Parenthesize` variants.